File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        l = 0
        r = max(len(word1), len(word2)) 
        res = []
        while l < r:
            if l < len((word1)):
                res.append(word1[l])
            if l < len((word2)):
                res.append(word2[l])
            l += 1

        return "".join(res)

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string ans;
        int l = 0;
        int r = max(word1.size(), word2.size());
        while (l < r) {
            if (l < word1.size()) {
                ans.push_back(word1[l]);
            }
            if (l < word2.size()) {
                ans.push_back(word2[l]);
            }
            l++;
        }
        return ans;
    }
};